### PR TITLE
Add AppConfig support

### PR DESCRIPTION
**Pull request by mgorski-mg**
## Description
Add AppConfig as another Systems Manager's source of configuration.

## Motivation and Context
It is extending the functionality for a new source of configuration.

## Testing
I created the ASP.NET app and configure it with AppConfig using this new extension.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] My code follows the code style of this project
- [ ] My change requires a change to the documentation
- [x] I have updated the documentation accordingly
- [x] I have read the **README** document
- [x] I have added tests to cover my changes
- [x] All new and existing tests passed

## License

- [x] I confirm that this pull request can be released under the Apache 2 license

[issues]: https://github.com/aws/aws-dotnet-extensions-configuration/issues
[license]: http://aws.amazon.com/apache2.0/
[cla]: http://en.wikipedia.org/wiki/Contributor_License_Agreement

### CODE REVIEW

**Review by normj — COMMENTED** [review]
I took a scan through the PR and it looks great. I need to do a deeper read through the code and try it out in action but wanted to give you some initial feedback that this looks like a great PR.

File: src/Amazon.Extensions.Configuration.SystemsManager/AppConfig/AppConfigProcessor.cs [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+        }
+
+        public async Task<IDictionary<string, string>> GetDataAsync()
+        {
+            var request = new GetConfigurationRequest
+            {
+                Application = Source.ApplicationId,
+                Environment = Source.EnvironmentId,
+                Configuration = Source.ConfigProfileId,
+                ClientId = Guid.NewGuid().ToString()
```
Comment (normj) [inline src/Amazon.Extensions.Configuration.SystemsManager/AppConfig/AppConfigProcessor.cs:46]: The Client ID should stay the same for the length of the process. This is going to report a different client id to AppConfig everytime there is a config reload.

File: src/Amazon.Extensions.Configuration.SystemsManager/AppConfig/AppConfigProcessor.cs [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+            using (var client = Source.AwsOptions.CreateServiceClient<IAmazonAppConfig>())
+            {
+                if (client is AmazonAppConfigClient impl)
+                {
+                    impl.BeforeRequestEvent += ServiceClientAppender.ServiceClientBeforeRequestEvent;
+                }
+
+                var response = await client.GetConfigurationAsync(request).ConfigureAwait(false);
+
+                return JsonConfigurationParser.Parse(response.Content);
```
Comment (normj) [inline src/Amazon.Extensions.Configuration.SystemsManager/AppConfig/AppConfigProcessor.cs:58]: ContentType needs to be checked to see if the content is JSON before parsing. If it is not JSON throw an error about it being an unsupported format.

File: src/Amazon.Extensions.Configuration.SystemsManager/AppConfig/AppConfigProcessor.cs [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+            if (source.EnvironmentId == null) throw new ArgumentNullException(nameof(source.EnvironmentId));
+            if (source.ConfigProfileId == null) throw new ArgumentNullException(nameof(source.ConfigProfileId));
+            if (source.AwsOptions == null) throw new ArgumentNullException(nameof(source.AwsOptions));
+            
+            Source = source;
+        }
+
+        public async Task<IDictionary<string, string>> GetDataAsync()
+        {
+            var request = new GetConfigurationRequest
```
Comment (normj) [inline src/Amazon.Extensions.Configuration.SystemsManager/AppConfig/AppConfigProcessor.cs:41]: AppConfig is highly recommending using the ClientConfigurationVersion field to reduce repeatedly fetching the configuration causing extra charges. https://docs.aws.amazon.com/appconfig/2019-10-09/APIReference/API_GetConfiguration.html#appconfig-GetConfiguration-request-ClientConfigurationVersion

**Review by normj — CHANGES_REQUESTED** [review]

File: src/Amazon.Extensions.Configuration.SystemsManager/AppConfig/AppConfigProcessor.cs (hunk 1 above)
Comment (mgorski-mg) [reply src/Amazon.Extensions.Configuration.SystemsManager/AppConfig/AppConfigProcessor.cs]: Good catch! Changed to be created once per app launch while creating Source.

File: src/Amazon.Extensions.Configuration.SystemsManager/AppConfig/AppConfigProcessor.cs (hunk 3 above)
Comment (mgorski-mg) [reply src/Amazon.Extensions.Configuration.SystemsManager/AppConfig/AppConfigProcessor.cs:44]: Fixed

File: src/Amazon.Extensions.Configuration.SystemsManager/AppConfig/AppConfigProcessor.cs (hunk 2 above)
Comment (mgorski-mg) [reply src/Amazon.Extensions.Configuration.SystemsManager/AppConfig/AppConfigProcessor.cs]: ContentType check added and NotImplementedException thrown for other ContentTypes than application/json.

**Comment by mgorski-mg** [comment]
@normj do you have a while to check the requested changes? :)

**Comment by wojciech-dabrowski** [comment]
@normj Any update on this? I'd love to use this functionality!

**Review by normj — APPROVED** [review]

**Comment by normj** [comment]
Got the PR merged into master. I'll ping back when it is released which is what I'm focused on now.

**Comment by normj** [comment]
Version 3.0.0 is released. Thanks so much for a cool feature and sorry for the delay getting it released.

**Pull request merged**